### PR TITLE
GPU Introduction

### LINKED ISSUE #11058: GPU Introduction
Hey guys, we're going to release the 3.1.3 ABI stable preview today. Can you write up a quick introduction in the #CategoryGPU section of SDL_gpu.h? Please include the goals, who it's for, where to find examples, etc.

### PR DESCRIPTION
**Pull request by thatcosmonaut**
Resolves #11058

### CODE REVIEW

**Comment by icculus** [comment]
Oh good, you added a computer section, thank you!

I'm good with this if @slouken approves.

File: include/SDL3/SDL_gpu.h [hunk 1]
```diff
@@ -24,7 +24,101 @@
...
+ * Then, dispatch compute work:
+ *
+ * - SDL_DispatchGPUCompute
+ *
+ * For advanced users, this opens up powerful GPU-driven workflows.
+ *
+ * This is an extremely quick overview that leaves out several important
+ * details. Already, though, one can see that GPU programming can be quite
+ * complex! If you just need simple 2D graphics, the
+ * [Render API](CategoryRender) is much easier to use but still
```
Comment (slouken) [inline include/SDL3/SDL_gpu.h:113]: Can you put a web link in here? This is in the header, which may be read off of the wiki.

**Comment by slouken** [comment]
Overall, looks good! Once you fix the link, feel free to give it one more look and merge!

**Comment by thatcosmonaut** [comment]
I'm going to reword a few things about render passes and also add a brief section about shaders.

**Comment by slouken** [comment]
> I'm going to reword a few things about render passes and also add a brief section about shaders.

Oh, good call.

**Comment by thatcosmonaut** [comment]
Made those changes if you want to give it another skim.

File: include/SDL3/SDL_gpu.h [hunk 2]
```diff
@@ -24,7 +24,118 @@
...
+ * For advanced users, this opens up powerful GPU-driven workflows.
+ *
+ * Graphics and compute pipelines require the use of shaders, which as mentioned above are small programs
+ * executed on the GPU. Each backend (Vulkan, Metal, D3D12) requires a different shader format.
+ * When the app creates the GPU device, the app lets the device know which shader formats the app can provide.
+ * It will then select the appropriate backend depending on the available shader formats and the backends available on the platform.
+ * When creating shaders, the app must provide the correct shader for the selected backend.
+ * If you would like to learn more about why the API works this way, there is a
+ * detailed [blog post](https://moonside.games/posts/layers-all-the-way-down/)
+ * explaining this situation. In the future, SDL may provide a universal shader format.
```
Comment (slouken) [inline include/SDL3/SDL_gpu.h:121]:
```suggestion
 * explaining this situation.
```

Let's not make any promises here. It's very possible, but I don't want people to hold their breath. :)

Comment (thatcosmonaut) [reply include/SDL3/SDL_gpu.h]: Good call.

**Comment by slouken** [comment]
I think this is good for the preview release. Feel free to squash and merge!

**Pull request merged**